### PR TITLE
New Sara Use Cases

### PR DESCRIPTION
**Pull request by b-quachtran**
New "Getting Started" Flows for Sara:
- `/get_started_intro` triggers the flow that will appear on the [Docs Intro Page](https://rasa.com/docs/rasa/)
This will offer options for:
    - playground - reporting a playground issue will enter it in a google sheet, like the sales contact data
    - install rasa
    - more info about rasa. triggers old onboarding flow - see below
- `/get_started_playground` triggers the flow that will appear on the [Rasa Playgrounds Page](https://rasa.com/docs/rasa/playground)

Modifications to what used to be the onboarding flow:
- choosing "more info about rasa" at start gives a `what is rasa` response then starts what used to be the onboarding flow ("new to rasa?" > "new to bot building?" etc.)
- the `how_to_get_started` intent was narrowed down to only actual requests to get started; nlu/core/rasa specific qs went to their respective intents (mostly faq), or enter_data
- the `how_to_get_started` intent also starts the new getting started flow in some contexts
- all `next_step` related stories & actions were removed, since installing rasa is covered by the new flow and rasa-init type stuff is covered by the playground
- `ActionGreetUser` is now only responsible for checking if the user has seen the privacy policy, and greeting the user.

### CODE REVIEW

**Comment by mvielkind** [comment]
Are any language models from `spacy` being loaded? When I start the action server with `rasa run actions --actions actions.actions` I get an error about not being able to load the `en` model in `spacy`. 

Here's the full error:

```
(rasa-demo) matthewvielkind@Matthews-MBP rasa-demo % rasa run actions --actions actions.actions
2021-01-05 08:50:22 INFO     rasa_sdk.endpoint  - Starting action endpoint server...
Traceback (most recent call last):
  File "/Users/matthewvielkind/environments/rasa-demo/bin/rasa", line 8, in <module>
    sys.exit(main())
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa/__main__.py", line 116, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa/cli/run.py", line 59, in run_actions
    sdk.main_from_args(args)
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa_sdk/__main__.py", line 21, in main_from_args
    args.auto_reload,
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa_sdk/endpoint.py", line 137, in run
    action_package_name, cors_origins=cors_origins, auto_reload=auto_reload
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa_sdk/endpoint.py", line 80, in create_app
    executor.register_package(action_package_name)
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa_sdk/executor.py", line 254, in register_package
    self._import_submodules(package)
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa_sdk/executor.py", line 210, in _import_submodules
    package = self._import_module(package)
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/rasa_sdk/executor.py", line 231, in _import_module
    module = importlib.import_module(name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/matthewvielkind/PycharmProjects/rasa-demo-pr-596/rasa-demo/actions/actions.py", line 21, in <module>
    from actions.api.algolia import AlgoliaAPI
  File "/Users/matthewvielkind/PycharmProjects/rasa-demo-pr-596/rasa-demo/actions/api/algolia.py", line 7, in <module>
    en_spacy = spacy.load("en")
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/spacy/__init__.py", line 30, in load
    return util.load_model(name, **overrides)
  File "/Users/matthewvielkind/environments/rasa-demo/lib/python3.7/site-packages/spacy/util.py", line 169, in load_model
    raise IOError(Errors.E050.format(name=name))
OSError: [E050] Can't find model 'en'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory.

```

File: data/stories/get_started_playground.yml
Comment (melindaloubser1) [inline data/stories/get_started_playground.yml:8]: This is an NLU problem as well as dialogue- where do we handle specific problem reports?
```
Your input ->  /get_started_playground                      
Do you have any questions or issues you're running into with Rasa Playground?
Your input ->  yes                                          
Ok, what can I help you with?
Your input ->  it's broken                                  
Can you describe the problem you're having?
Your input ->  it's not training                            
Ok, what can I help you with?
Your input ->  the playground, it's not training my model   
? Sorry, I'm not sure I've understood you correctly 🤔 Do yo
u mean...  (Use arrow keys)                                 
 » 1: nlu_fallback (/nlu_fallback{"product": "playground"})
   2: Something else (/out_of_scope)
   Type out your own message...    
 ```

**Review by melindaloubser1 — CHANGES_REQUESTED** [review]
I think the playground flow needs a bit of fleshing out - what happens when users report specific problems? Do we always send them to the forum regardless?

**Comment by mvielkind** [comment]
Thanks @melindaloubser1 ! I see it there in the `Dockerfile` now. The current setup doesn't show the user how they can download the `spacy` model and build the shortcut link, which will cause an error when you start the action server. I know you can add a line of code to download the model from `spacy` in `requirements.txt`, but I'm not sure you can make the shortcut for the model `en` the code uses.

**Comment by melindaloubser1** [comment]
you're right, can't include the model download in the requirements directly.  We should add a note on that in the readme and maybe an `install` recipe in the Makefile.

**Comment by mvielkind** [comment]
Agreed. I'll test it out to confirm that will work and add another commit with that proposed change.

**Comment by b-quachtran** [comment]
@melindaloubser1 For now I think it makes sense to send them to the forum regardless and use that as a starting point to handle frequent problems through a different mechanism after the bot is live. Wdyt?

**Comment by melindaloubser1** [comment]
@b-quachtran Ja it does, we should add some examples of things they might report though, or everything will be gotten as "nlu_fallback". Dumb q, where is the story/rule that actually sends them to the forum after they report?

**Comment by b-quachtran** [comment]
@melindaloubser1 Def not a dumb question considering I didn't really build out that broken flow correctly 😅 . It's in data/stories/broken.yml now and let me know what can be improved there.

**Comment by melindaloubser1** [comment]
@b-quachtran do you also see it not registering `utter_chitchat` from the responses file? 
```
UserWarning: Action 'utter_chitchat' is listed as a response action in the domain file, but there is no matching response defined. Please check your domain.
```

**Comment by b-quachtran** [comment]
@melindaloubser1 Yeah, apparently it's a known bug at the moment: https://github.com/RasaHQ/rasa/issues/7645

**Comment by melindaloubser1** [comment]
@b-quachtran Sorry for taking so long to come back to this. I'm thinking this might not be a very good user experience:

image
Can we change the response to the forum redirect? something like:

> I've made a note of the problem you're experiencing! In the meantime, if you're still stuck, please ask for help on our forum.

**Comment by melindaloubser1** [comment]
Another question: Should there not be an intermediate between saying "yes, I want to try the playground" and asking if they need help with the playground? Some kind of a plug or description? e.g.
> utter_playground_intro:
   - text: In the [Rasa Playground](link to playground just in case) you can  follow the instructions to build a Rasa prototype entirely online, without having to install Rasa locally. Once you've built your prototype, you can talk to your bot online and download your project to continue developing it locally

File: data/stories/get_started_intro.yml
Comment (melindaloubser1) [inline data/stories/get_started_intro.yml:11]: I think we need to follow a similar flow also for `how_to_get_started`; e.g. things like `try it out` in response to  `utter_ask_playground_or_install` currently will start the old get started flow. We could split up that intent into `get_started_intro` (e.g. try it out, how do i build a bot, etc) and `install_rasa` (and maybe keep the rest where it is?) so that there's less confusion between which getting started flow the user follows

**Review by melindaloubser1 — CHANGES_REQUESTED** [review]
Couple of notes on responses and I think we need to do something about `how_to_get_started` and the related stories.

**Comment by b-quachtran** [comment]
@melindaloubser1 Sorry for the super late follow-up, would you have any time to take on some of these flow updates in the short term? I don't think I'll have a chance to work on this for another week or so.

**Comment by melindaloubser1** [comment]
@b-quachtran I have some time for it now

**Comment by melindaloubser1** [comment]
@b-quachtran finally got round to resolving the conflicts I had after adding the 3rd option and merging the how_to_get_started and get_started_intro flows. I can't officially request your review because you opened this PR, but can you review my changes?

File: config.yml
Comment (melindaloubser1) [inline config.yml:49]: changed order of policies so that if there are rule conflicts they will be determined before ted policy trains to save time

**Comment by melindaloubser1** [comment]
@erohmensing Since you know what the flow was before pretty well, it would be great if you could take a look

**Comment by melindaloubser1** [comment]
Compared to previous results here https://github.com/RasaHQ/rasa-demo/pull/599
Some things have gone down.

**Comment by erohmensing** [comment]
@melindaloubser1 When I made the training on NLU testing, i recommended that they remove the FallbackClassifier from the pipeline when running NLU tests. Wdyt about adding a separate config for that without the FallbackClassifier? Then we should be able to actually interpret the `confused_with` etc and properly set fallback thresholds. 

(The NLU testing scripts should probably just ignore it by default, but havent gotten around to raising that convo yet) 

Re: reviewing, does Friday work? Up until then i'll be mostly focused on the Rasa X release

**Comment by melindaloubser1** [comment]
Good idea! Yes that's fine, still need @b-quachtran 's review too.

**Review by melindaloubser1 — APPROVED** [review]
I'm approving but since I took over editing this PR @ArjaanBuijk I'll wait on your review before merging

File: .github/workflows/continuous_integration.yml
Comment (ArjaanBuijk) [inline .github/workflows/continuous_integration.yml:131]: Great idea indeed !

**Comment by ArjaanBuijk** [comment]
@melindaloubser1 ,
I am trying to train the model, but am getting an error:

```bash
ActionNotFoundException: Cannot access action 'action_next_step', as that name is not a registered action for this domain.
```

Was something not pushed to the repo?

--
Also, I noticed that the Type Tests fail due to networkx module. I already fixed that in a branch that we will merge to main tomorrow, so that will be addressed.

**Comment by melindaloubser1** [comment]
aw man I thought I fixed that. I think the merge of main may have put some next_step stories back, I'll check.

**Comment by melindaloubser1** [comment]
@arjaan fixed; rasa data validate no longer complains

**Pull request merged**
